docs/branches.md

```diff
@@ -52,6 +52,26 @@ some remote.
 If you want to know the internals of branch tracking, consult the 
 [Design Doc][design].
 
+### Terminology summary
+
+- A **remote branch** is a branch ref on the remote. `jj` can find out its state
+  only when it's actively communicating with the remote. 
+- A **tracked (remote) branch** is defined above. You can make a remote branch
+  tracked with the [`jj branch track` command](#manually-tracking-a-branch), for
+  example.
+- A **local tracking branch** is the local branch that `jj` tries to keep in
+  sync with the tracked remote branch. For example, after `jj branch track
+  develop@origin`, there will be a local branch `develop` that's tracking the
+  remote `develop@origin` branch.
+- A **remote-tracking branch** is a term common in the Git world, but `jj` has
```

Comment:
I would remove "remote-tracking branch" from the list because it's confusing. Perhaps, it can be merged into the "remote branch" definition.

We won't need to distinguish the remote branches in the local repo from the local branches in the remote repo in many contexts.

Reply:
Done. I still discuss the distinction, but moving it into the same bullet point probably deepmhasizes it.